src/SFA.DAS.Reservations.Data/Repository/ReservationIndexRepository.cs

```diff
@@ -124,5 +223,12 @@ private class IndexRegistryEntry
             public string Name { get; set; }
             public DateTime DateCreated { get; set; }
         }
+
+        private struct FilterValues
+        {
+            public ICollection<string> Courses { get; set; }
+            public ICollection<string> AccountLegalEntityNames { get; set; }
+            public ICollection<string> StartDates { get; set; }
+        }
     }
 }
```

Comment:
We need to look at moving out the logic where it builds up the elastic queries, I think this would be better if it came from the Domain model instead and was determined on the values supplied to that. Would make the repository a bit cleaner. Lets create a Task to tackle this